native/cocos/editor-support/spine-wasm/spine-type-export.cpp

```diff
@@ -85,20 +34,467 @@ using SPVectorPathConstraintPtr = Vector<PathConstraint*>;
 using SPVectorTimelinePtr = Vector<Timeline*>;
 using SPVectorTrackEntryPtr = Vector<TrackEntry*>;
 using SPVectorUpdatablePtr = Vector<Updatable*>;
+using SPVectorSkinEntryPtr = Vector<Skin::AttachmentMap::Entry*>;
+using SPVectorVectorSkinEntryPtr = Vector<SPVectorSkinEntryPtr>;
+using SPVectorDebugShape = Vector<SpineDebugShape>;
+
+#define DEFINE_ALLOW_RAW_POINTER(type) \
+namespace emscripten { namespace internal { \
+    template<> \
+    struct TypeID<type*> { \
+        static constexpr TYPEID get() { \
+            return TypeID<type>::get(); \
+        } \
+    }; \
+    template<> \
+    struct TypeID<const type*> { \
+        static constexpr TYPEID get() { \
+            return TypeID<type>::get(); \
+        } \
+    }; \
+}}
+
+
+#define DEFINE_SPINE_CLASS_TYPEID(cls) \
+namespace emscripten { namespace internal { \
+    template<> \
+    constexpr TYPEID getLightTypeID<cls>(const cls& value) { \
+        return value.getRTTI().getClassName(); \
```

Comment:
Spine runtime has already supported `RTTI` functionality, so we disable which in standard c++ by `-fno-exception` in CMakeLists.txt. This could save about 20 KB wasm file size (uncompressed).

Reply:
To use our own version of TypeID system, emscripten provide some internal APIs whose we could override here.